### PR TITLE
feat: Add reports class

### PR DESCRIPTION
**Pull request by Jacksonball-jlb**
In the progress of working on issue #25. Only have a basic reports template. Going to add a lot more stuff but feel free to give critiques if need be.

### CODE REVIEW

File: reports.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            # Name and total hours:minutes
+            c.setFont("Helvetica", 12)
+            c.drawString(80, y_pos, name)
+            c.drawString(370, y_pos, time)
+            y_pos -= 16
+
+        # Save the PDF
+        c.save()
+
+    def create_detailed_pdf(self, file_name, company_name, title, username,
```
Comment (AungA07) [inline reports.py:82]: rename this method to something like "create_user_specific_pdf"

File: reports.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    def __init__(self):
+        """Placeholder. Need help to put stuff here"""
+
+    def calculate_pages(self):
+        """
+        Simulated for loop counter (for total page count)
+        """
+        fake_y_pos = 615
+        total_pages = 1
+        for i in range(len(data['names'])):
```
Comment (AungA07) [inline reports.py:16]: This should be an argument instead

File: reports.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            if fake_y_pos < 40:
+                total_pages += 1
+                fake_y_pos = 730
+            fake_y_pos -= 16
+        return total_pages
+
+    def create_basic_pdf(self, file_name, company_name, title, start_date, end_date, data):
```
Comment (AungA07) [inline reports.py:23]: Data would be better off as an inititalized attribute

File: reports.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        total_pages = self.calculate_pages()
+
+        # First page set up
+        c.setFont("Helvetica", 10)
+        c.drawString(520, 30, f"Page {page_num} of {total_pages}")
+        c.drawString(30, 30, current_time)
+
+        # Loop through all names and total minutes
+        for i, (date, clock_in, clock_out, total_time) in enumerate(
+                zip(data['date'], data['clock_in'], data['clock_out'], data['times'])):
```
Comment (AungA07) [inline reports.py:128]: these should be accessed through the database

File: reports.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            c.drawString(75, y_pos, date)
+            c.drawRightString(245, y_pos, clock_in)
+            c.drawRightString(375, y_pos, clock_out)
+            c.drawString(460, y_pos, total_time)
+            y_pos -= 16
+
+        # Save the PDF
+        c.save()
+
+# Hardcode info for now
```
Comment (AungA07) [inline reports.py:151]: eventually get rid of this testing code at the end of the file

**Review by AungA07 — CHANGES_REQUESTED** [review]

File: reports.py (hunk 2 above)
Comment (Jacksonball-jlb) [reply reports.py]: This is just a counter. There might be a better way but I haven't found it. Will try to refactor as soon as I can.

File: reports.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    @staticmethod
+    def calculate_number_of_pages(data):
+        """
+        Calculating number of pages the report will have
+        """
+        items_per_page = (615 - 40) // 16
+        total_pages = (len(data) + items_per_page - 1) // items_per_page
+        return total_pages + 1
+
+    def create_basic_pdf(self, file_name, company_name, title, start_date, end_date):
```
Comment (AungA07) [inline reports.py:21]: rename this function to something more descriptive of the actual function (like create_all_user_pdf)

File: reports.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+            c.drawString(75, y_pos, row[1])
+            c.drawRightString(245, y_pos, row[2])
+            c.drawRightString(375, y_pos, row[3])
+            c.drawString(460, y_pos, row[4])
+            y_pos -= 16
+
+        # Save the PDF
+        c.save()
+
+    def download_pdf(self, file_name):
```
Comment (AungA07) [inline reports.py:151]: add "print" argument and a comment saying it can only run on windows
os.startfile(file_name, "print")

File: reports.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+                            start_date, end_date, name, student_id):
+        # Initialize necessary variables
+        current_time = datetime.now()
+        current_time = current_time.strftime("%m/%d/%Y %I:%M:%S %p")
+        page_num = 1
+        y_pos = 605
+        data = self.access.admin_read_self_table(student_id)
+
+        # Set up the canvas
+        c = canvas.Canvas(file_name, pagesize=letter)
```
Comment (AungA07) [inline reports.py:92]: same here thing here: create a folder for the file to be stored in

**Review by AungA07 — APPROVED** [review]
Looks good now!

**Review by LoganGhast — APPROVED** [review]

**Pull request merged**